azure-pipelines.yml

```diff
@@ -116,6 +116,87 @@ stages:
           flake8 --select F401 great_expectations tests
         name: UnusedImportChecker
 
+  - stage: docker_tests
+    dependsOn: []
+    pool:
+      vmImage: 'ubuntu-latest'
+    jobs:
+      - job: build_push_test_
+        timeoutInMinutes: 120
+        strategy:
+          matrix:
+            Python37:
+              minor_version: '7'
+            Python38:
+              minor_version: '8'
+            Python39:
+              minor_version: '9'
+        services:
+          postgres: postgres
+        variables:
+          DOCKER_TAG: none
+        steps:
+          - bash: |
+              # login to docker first so we die fast if this fails
+              docker login -u ${DOCKER_USER} -p ${DOCKER_TOKEN}
+
+              EPOCH=`date +%s`
+              TAG="3.$(minor_version)_develop_${EPOCH}_${RANDOM}"
+              # We repull the develop branch. We could build the local branch that is already present but we should
+              # prune it's depth since we don't need the git history.
+              # CMD="docker buildx build -f docker/Dockerfile.tests --tag greatexpectations/test:${TAG} --target azure --build-arg PYTHON_VERSION=3.$(minor_version) --build-arg SOURCE=github --build-arg BRANCH=develop ."
+              # Temporarily use local copy of repo
+              CMD="docker buildx build -f docker/Dockerfile.tests --tag greatexpectations/test:${TAG} --target azure --build-arg PYTHON_VERSION=3.$(minor_version) ."
+              echo ${CMD}
+              eval ${CMD}
+              PUSH="docker push greatexpectations/test:${TAG}"
+              echo ${PUSH}
+              eval ${PUSH}
+              echo "Docker image can be found at https://hub.docker.com/r/greatexpectations/test/tags"
```

Comment:
Wanted to include some explicit logging lines to let people who are debugging issue know where they can get the docker image they want.